hs_geographic_feature_resource/models.py

```diff
@@ -112,26 +289,31 @@ class Meta:
 processor_for(GeographicFeatureResource)(resource_processor)
 
 
-# define the GeographicFeatureMetaData metadata
-class GeographicFeatureMetaData(CoreMetaData):
-    geometryinformation = GenericRelation(GeometryInformation)
-    fieldinformation = GenericRelation(FieldInformation)
-    originalcoverage = GenericRelation(OriginalCoverage)
-    originalfileinfo = GenericRelation(OriginalFileInfo)
+class GeographicFeatureMetaDataMixin(models.Model):
+    """This class must be the first class in the multi-inheritance list of classes"""
+    geometryinformations = GenericRelation(GeometryInformation)
+    fieldinformations = GenericRelation(FieldInformation)
+    originalcoverages = GenericRelation(OriginalCoverage)
+
+    class Meta:
+        abstract = True
 
     @property
-    def resource(self):
-        return GeographicFeatureResource.objects.filter(object_id=self.id).first()
+    def geometryinformation(self):
+        return self.geometryinformations.all().first()
+
+    @property
+    def originalcoverage(self):
+        return self.originalcoverages.all().first()
 
     @classmethod
     def get_supported_element_names(cls):
         # get the names of all core metadata elements
-        elements = super(GeographicFeatureMetaData, cls).get_supported_element_names()
+        elements = super(GeographicFeatureMetaDataMixin, cls).get_supported_element_names()
```

Comment:
@pkdash Now that ```GeographicFeatureMetaDataMixin``` class is a subclass of ```models.Model```, get_supported_element_names() don't exist in the parent/super class any more, right? I cannot figure out the benefit of changing GeographicFeatureMetaData to GeographicFeatureMetaDataMixin at this point, perhaps when I review further, I'll get it. But it'd be good if you can explain a bit on the thinking behind this change. Is it because the Mixin class will be inherited by both geo feature resource type and geo feature file type in composite resources?